### PR TITLE
Fix the new message button doesn't work on some comment linkings

### PR DESCRIPTION
**Pull request by tsa321**
### Fixed Issues

$ https://github.com/Expensify/App/issues/45004
PROPOSAL: https://github.com/Expensify/App/issues/45004#issuecomment-2254297005

### Tests

1. Go to a chat with 200 - 210 messages
2. Copy the link of the 70th comment and send it
3. Copy the link of the 140th comment and send it
4. Go to troubleshoot and clear cache and restart
5. Click on the first link sent on step 2
6. Click on the "New message" badge that appeared at the top of the chat
7. Confirm the page scrolls to the bottom of the comment history
8. Click on the second link sent on step 3.
9. Click on the "New message" badge once again
10. Confirm the page scrolls to the bottom of the comment history

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Go to a chat with 200 - 210 messages
2. Copy the link of the 70th comment and send it
3. Copy the link of the 140th comment and send it
4. Go to troubleshoot and clear cache and restart
5. Click on the first link sent on step 2
6. Click on the "New message" badge that appeared at the top of the chat
7. Confirm the page scrolls to the bottom of the comment history
8. Click on the second link sent on step 3.
9. Click on the "New message" badge once again
10. Confirm the page scrolls to the bottom of the comment history
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/c20c9c95-8a94-4daf-ace2-2e39212814a8

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/98b049de-2e75-4aa2-af93-eaa9de31c2a5

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/9045b4ca-3705-4cfe-bede-a9a898659288

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/52af8eef-4263-4c8b-bce4-c3c8461d8025

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/49f4dc14-0a93-4d24-84fb-8a602e8afe45

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/ad82962b-6a42-4979-8d6f-4ee3556f82c0

</details>

### CODE REVIEW

**Comment by s77rt** [comment]
@tsa321 Is the pagination bug still reproducible? If so can you please add reproduction steps?

**Comment by s77rt** [comment]
@tsa321 It seems that something broke


https://github.com/user-attachments/assets/2a31b146-1712-422c-aba9-900cd09d99d6

**Comment by s77rt** [comment]
The latest commit didn't fix the reported but and now the "New message" button is not visible


https://github.com/user-attachments/assets/a7db08cd-23d0-464d-93f2-485fdde86e2a

**Comment by s77rt** [comment]
@tsa321 Those bugs appear to be on `main` as well and must be fixed as they block testing the original bug

File: src/pages/home/ReportScreen.tsx [hunk 1]
```diff
@@ -95,7 +96,8 @@ function getParentReportAction(parentReportActions: OnyxEntry<OnyxTypes.ReportAc
     return parentReportActions[parentReportActionID ?? '0'];
 }
 
-function ReportScreen({route, currentReportID = '', navigation}: ReportScreenProps) {
+function ReportScreen({currentReportID = '', navigation}: ReportScreenProps) {
+    const route = useRoute<RouteProp<AuthScreensParamList, typeof SCREENS.REPORT>>();
```
Comment (s77rt) [inline src/pages/home/ReportScreen.tsx:100]: Why is this change necessary?

File: src/libs/PaginationUtils.ts [hunk 2]
```diff
@@ -124,7 +124,7 @@ function mergeAndSortContinuousPages<TResource>(sortedItems: TResource[], pages:
     const result = [sortedPages[0]];
     for (let i = 1; i < sortedPages.length; i++) {
         const page = sortedPages[i];
-        const prevPage = sortedPages[i - 1];
+        const prevPage = result?.at(-1) ?? sortedPages[i - 1];
```
Comment (s77rt) [inline src/libs/PaginationUtils.ts:127]: Why do we fallback to `sortedPages[i - 1]`?

**Review by s77rt — CHANGES_REQUESTED** [review]

File: src/pages/home/ReportScreen.tsx (hunk 1 above)
Comment (s77rt) [reply src/pages/home/ReportScreen.tsx:100]: But this is already working on `main`, what broke it?

https://github.com/user-attachments/assets/06481aa5-ab8d-43ba-b792-da97bc5b42a8

File: src/libs/PaginationUtils.ts (hunk 2 above)
Comment (s77rt) [reply src/libs/PaginationUtils.ts:127]: `prevPage` is expected to be `undefined` at first, no need for the fallback

File: src/libs/PaginationUtils.ts [hunk 3]
```diff
@@ -124,7 +124,7 @@ function mergeAndSortContinuousPages<TResource>(sortedItems: TResource[], pages:
     const result = [sortedPages[0]];
     for (let i = 1; i < sortedPages.length; i++) {
         const page = sortedPages[i];
-        const prevPage = sortedPages[i - 1];
+        const prevPage = result[result.length - 1];
```
Comment (s77rt) [inline src/libs/PaginationUtils.ts:127]: prepage (`result[result.length - 1]`) could be same as page if the result contains only one page

File: src/pages/home/ReportScreen.tsx (hunk 1 above)
Comment (s77rt) [reply src/pages/home/ReportScreen.tsx:100]: @tsa321 Can you please clarify the solution independently for those two bugs:
- Not showing "new message" button
- Not navigating on clicking same report action link


(Also please merge `main`)

**Review by s77rt — CHANGES_REQUESTED** [review]

File: src/libs/PaginationUtils.ts (hunk 3 above)
Comment (s77rt) [reply src/libs/PaginationUtils.ts:127]: If result length is 1, then:
`page = sortedPages[0]`
and
`prevPage = sortedPages[0]`

page is same as prevPage which is misleading and could cause bugs

Comment (s77rt) [reply src/libs/PaginationUtils.ts:127]: Ah right! my bad

**Comment by s77rt** [comment]
Can you check the following bug, after clicking on the message link, the message is not in view

https://github.com/user-attachments/assets/38f8a535-1bca-499f-a8dd-f3ff4d6be1df

**Comment by s77rt** [comment]
@tsa321 It would be better if we can get this fixed or at least find what went wrong. Can't you reproduce the bug on `main`? There are no specific steps besides clicking on the report action link

**Comment by s77rt** [comment]
@tsa321 Let's not block on that.

> Click on the second link sent on step 5.

This should refer to step 3

**Comment by s77rt** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



https://github.com/user-attachments/assets/15c9c15a-f6d4-40af-ae30-af12fe0480b9


</details>

<details>
<summary>iOS: Native</summary>



https://github.com/user-attachments/assets/2b9c9d04-b83d-4f80-911f-299df5caca6f


</details>

<details>
<summary>iOS: mWeb Safari</summary>



https://github.com/user-attachments/assets/4371d743-25ff-4a9c-afe0-d028e236e5c7


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



https://github.com/user-attachments/assets/4c723c56-a86f-41d1-91b4-b6810fd5a3e2


</details>

<details>
<summary>MacOS: Desktop</summary>



https://github.com/user-attachments/assets/31f98b2f-82cd-436c-99ac-e83c1576e33f


</details>

**Comment by s77rt** [comment]
On Android the new message button does not scroll to the bottom

https://github.com/user-attachments/assets/44abf2ef-09ea-4b8e-a921-b637a6a80030

**Comment by s77rt** [comment]
@tsa321 That code was there for a while, it can't be the culprit. Can you try find the offending PR or the root cause. Also try disable strict mode and test on production

**Comment by s77rt** [comment]
@tsa321 Before going with a solution, we need to find the root cause and since this broke recently it's better to find the offending PR first

**Comment by s77rt** [comment]
@tsa321 Can you revert the latest changes and merge `main`. I think it would be better to have this handled separately

**Review by s77rt — APPROVED** [review]
on Android it's partially working as expected but that seems due to another cause. Let's not block on that


https://github.com/user-attachments/assets/83af3543-86e9-4f9a-8adb-f5548669ed65

**Review by francoisl — APPROVED** [review]

**Comment by Beamanator** [comment]
FYI I believe this was deployed to prod yesterday, from this checklist - https://github.com/Expensify/App/issues/47219

**Pull request merged**